Sources/Commands/PackageTools/DumpCommands.swift

```diff
@@ -69,12 +69,22 @@ struct DumpSymbolGraph: SwiftCommand {
         let targets = try buildSystem.getPackageGraph().rootPackages.flatMap{ $0.targets }.filter{ $0.type == .library || $0.type == .executable }
         for target in targets {
             print("-- Emitting symbol graph for", target.name)
-            try symbolGraphExtractor.extractSymbolGraph(
+            let result = try symbolGraphExtractor.extractSymbolGraph(
                 target: target,
                 buildPlan: buildPlan,
+                outputRedirection: .collect(redirectStderr: false),
                 outputDirectory: symbolGraphDirectory,
                 verboseOutput: swiftTool.logLevel <= .info
             )
+
+            if result.exitStatus != .terminated(code: 0) {
+                switch result.stderrOutput {
+                case .success(let value):
+                    try FileHandle.standardError.write(contentsOf: "Failed: \(String(decoding: value, as: UTF8.self))".data(using: .utf8)!)
+                case .failure(let error):
+                    try FileHandle.standardError.write(contentsOf: "Internal error: \(error)".data(using: .utf8)!)
+                }
+            }
```

Comment:
IIRC tools always just use "print" directly

Reply:
Kind of, right now this command emits a little bit of output through `print` and a lot of other output onto stderr (the build part). It seemed more appropriate to have these show up on stderr.

Reply:
IIRC the idea was that the command itself always outputs to stdout, while any side effects (eg build output for commands that need to build but are not "swift-build") go to stderr. this is the way all commands are implemented afaik. is there a reason to deviate in this case. 

also, in other places we use `TSCBasic.stderrStream` to access stderr, so may want to use that just to be consistent

Reply:
It just seems completely wrong to send errors to stdout. But I'm fine with changing it for consistency.

Reply:
so looks like most other tools just throw an error and that bubbles up to arg parser which does this:

```
public static func exit(
    withError error: Error? = nil
  ) -> Never {
    guard let error = error else {
      Platform.exit(ExitCode.success.rawValue)
    }
    
    let messageInfo = MessageInfo(error: error, type: self)
    let fullText = messageInfo.fullText(for: self)
    if !fullText.isEmpty {
      if messageInfo.shouldExitCleanly {
        print(fullText)
      } else {
        print(fullText, to: &Platform.standardError)
      }
    }
    Platform.exit(messageInfo.exitCode.rawValue)
  }
```

so looks like it will eventually print to stderr when the underlying too throws. perhaps we should just do that? otherwise stderr seems correct

Reply:
Throwing doesn't seem viable, because of the unstructured multi-line output we get in the error case, the output will be very unreadable. I opted instead for emitting errors through the observability scope.